### PR TITLE
Decrease resume session complexity

### PR DESCRIPTION
**Pull request by muneebahmed10**
*Description*:
- Refactor Resume session to decrease complexity
- Update MQTTAgent_ProcessLoop

### CODE REVIEW

**Review by leegeth — APPROVED** [review]

File: source/mqtt_agent.c [hunk 1]
```diff
@@ -200,6 +200,24 @@ static MQTTStatus_t createAndAddCommand( CommandType_t commandType,
...
+ * @param[in] pMqttAgentContext Agent context for the MQTT connection.
+ *
+ * @return MQTTSuccess if all publishes resent successfully, else error code
+ * from #MQTT_Publish.
+ */
+static MQTTStatus_t resendPublishes( MQTTAgentContext_t * pMqttAgentContext );
+
+/**
+ * @brief Clear the list of pending acknowledgments by invoking each callback
+ * with #MQTTBadParameter.
```
Comment (dan4thewin) [inline source/mqtt_agent.c:215]: Using MQTTBadParameter strikes me as odd.  Would MQTTRecvFailed be a better choice?

Comment (muneebahmed10) [reply source/mqtt_agent.c:215]: Good catch. I actually used MQTTBadResponse but documented it incorrectly. Should I change them both to MQTTRecvFailed or keep as MQTTBadResponse?

Comment (dan4thewin) [reply source/mqtt_agent.c:215]: I don't think there is a best answer.  My first thought for bad response would be "I got something and it was bogus."  The receive failed makes me think "I called socket read and got an error." which is still not quite right, but is a better fit for a lost connection.  What do you think?

**Review by dan4thewin — APPROVED** [review]

File: source/mqtt_agent.c (hunk 1 above)
Comment (muneebahmed10) [reply source/mqtt_agent.c]: I think best would be if this situation had its own error code, but given the current implementation decision of using coreMQTT's codes, I agree with you that receive failed is the closest fit.

**Pull request merged**